### PR TITLE
[lint] Add golangci-lint automation + fix existing issues

### PR DESCRIPTION
**Pull request by tomaszjonak**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -226,8 +229,8 @@ func main() {
 			*port = "8001"
 		}
 		log.Infof("listen on port: %s", *port)
-		log.Fatal(r.Run(":" + *port))
+		log.Error(r.Run(":" + *port))
 	default:
-		log.Fatalf("unsupported mode %s", *mode)
```
Comment (noboruma) [inline main.go]: Let's keep them as `Fatal` to get an exit error that is not 0

File: scanner/controller/vulnerability-scan-service/cveservice.go [hunk 2]
```diff
@@ -99,44 +96,27 @@ func (c *VulnerabilityScanService) Scan(gc *gin.Context) {
...
-		}
+		// if len(report) > 0 {
+		// finalReportInByte := format(report)
+		// err = publisher.PublishDocument(IngestAPI, bytes.NewReader(finalReportInByte))
+		// if err != nil {
+		// 	publisher.PublishScanError(err.Error())
+		// 	log.Errorf("error on PublishDocument: %s", err.Error())
+		// 	return
+		// }
+		// }
```
Comment (noboruma) [inline scanner/controller/vulnerability-scan-service/cveservice.go]: Let's remove that entirely

**Review by noboruma — APPROVED** [review]
Good to go after 2 comments addressed

File: main.go (hunk 1 above)
Comment (tomaszjonak) [reply main.go]: Gonna move to `Panicf`. As we discussed Fatal triggers `os.Exit` which doesn't run defers.

**Pull request merged**